modules/dockermodelrunner/dockermodelrunner.go

```diff
@@ -0,0 +1,132 @@
+	"github.com/testcontainers/testcontainers-go/modules/socat"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const (
+	modelRunnerEntrypoint = "model-runner.docker.internal"
+	modelRunnerPort       = 80
+)
+
+// Container represents the DockerModelRunner container type used in the module
+type Container struct {
+	*socat.Container
+	model   string
+	baseURL string
+}
+
+// Run creates an instance of the DockerModelRunner container type.
+func Run(ctx context.Context, opts ...testcontainers.ContainerCustomizer) (*Container, error) {
+	return RunWithImage(ctx, socat.DefaultImage, opts...)
+}
+
+// RunWithImage creates an instance of the DockerModelRunner container type, using a specific image for the socat container.
+func RunWithImage(ctx context.Context, img string, opts ...testcontainers.ContainerCustomizer) (*Container, error) {
+	settings := defaultOptions()
+
+	// Process model runner options.
+	for _, opt := range opts {
+		if apply, ok := opt.(Option); ok {
+			if err := apply(&settings); err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	// Add socat options, which are applied to the socat container.
+	opts = append(opts, testcontainers.WithWaitStrategy(
+		wait.ForListeningPort("80/tcp"),
+		wait.ForHTTP("/").WithPort("80/tcp").WithStatusCodeMatcher(func(status int) bool {
+			return status == http.StatusOK
+		}).WithResponseMatcher(func(body io.Reader) bool {
+			bodyBytes, err := io.ReadAll(body)
+			if err != nil {
+				return false
+			}
+			return strings.Contains(string(bodyBytes), "The service is running")
+		}),
+	))
+	opts = append(opts, socat.WithTarget(socat.NewTarget(modelRunnerPort, modelRunnerEntrypoint)))
+
+	socatCtr, err := socat.Run(ctx, img, opts...)
+	var c *Container
+	if socatCtr != nil {
+		c = &Container{Container: socatCtr, model: settings.model}
+	}
+
+	if err != nil {
+		return c, fmt.Errorf("socat run: %w", err)
+	}
+
+	c.baseURL = socatCtr.TargetURL(modelRunnerPort).String()
+
+	if settings.model != "" {
+		err := c.PullModel(ctx, settings.model)
+		if err != nil {
+			return c, fmt.Errorf("pull model: %w", err)
+		}
+	}
+
+	return c, nil
+}
+
+// InspectModel returns a model that is already pulled using the Docker Model Runner format.
+// The name of the model is in the format of <name>:<tag>.
+func (c *Container) InspectModel(ctx context.Context, namespace string, name string) (*types.ModelResponse, error) {
+	dmrClient := client.NewClient(c.baseURL)
+
+	modelResponse, err := dmrClient.InspectModel(ctx, namespace, name)
+	if err != nil {
+		return nil, fmt.Errorf("inspect model: %w", err)
+	}
+
+	return modelResponse, nil
+}
+
+// ListModels lists all models that are already pulled using the Docker Model Runner format.
+func (c *Container) ListModels(ctx context.Context) ([]types.ModelResponse, error) {
+	dmrClient := client.NewClient(c.baseURL)
+
+	models, err := dmrClient.ListModels(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("list models: %w", err)
+	}
+
+	return models, nil
+}
```

Comment:
tc modules must not be coupled to client SDKs

Reply:
Could you clarify the intent of that guidance? 

Are you suggesting that all callers have to take an endpoint from testcontainers create a client and then make the ListModels call?

Also does the fact this is a provided client not an external implementation have any bearing?

Reply:
The drmClient is an http client provided by the module. We could, as follow up, allow client code to customize the client, but with the current implementation, the client code cannot change it, on purpose.